### PR TITLE
Fixed a TemporaryFile crash

### PR DESCRIPTION
**Pull request by czechboy0**
Fixed a TemporaryFile crash caused by higher ASCII characters in the directory path.

### CODE REVIEW

**Comment by czechboy0** [comment]
@swift-ci smoke test

File: Tests/BasicTests/TemporaryFileTests.swift [hunk 1]
```diff
@@ -84,6 +84,17 @@ class TemporaryFileTests: XCTestCase {
...
+    
+    func testNonStandardASCIIName() throws {
+        let dir = try determineTempDirectory().appending(component: "Héllo")
+        try localFileSystem.createDirectory(dir)
+        defer {
+            try? localFileSystem.removeFileTree(dir)
+        }
+        try withTemporaryFile(dir: dir) { file in
+            XCTAssertTrue(localFileSystem.isFile(file.path))
+        }
```
Comment (aciidb0mb3r) [inline Tests/BasicTests/TemporaryFileTests.swift]: nitpick: You can simplify the test to:

```
try withTemporaryFile(prefix: "Héllo") { file in
    XCTAssertTrue(localFileSystem.isFile(file.path))
}
```

Comment (czechboy0) [reply Tests/BasicTests/TemporaryFileTests.swift]: Ah good point, let me fix it up.

**Comment by czechboy0** [comment]
@swift-ci smoke test

**Pull request merged**